routes/eth.go

```diff
@@ -367,14 +375,121 @@ func (fes *APIServer) QueryETHRPC(ww http.ResponseWriter, req *http.Request) {
 	}
 }
 
+type MetamaskSignInRequest struct {
+	AmountNanos uint64
+	Signer      []byte
```

Comment:
Maybe let's change `Signer` to `UserPublicKeyBase58Check` and pass it in base58check string instead of a byte array? Can we also add some comments here to explain the params?

Reply:
Is there a benefit to doing that? Having signer, message, and signature all as byte arrays would make the request feel more uniform. We would also need to convert it back down to a byte array we deriving the eth public address anyways.

Reply:
It displays better and feels more intuitive to pass strings rather than number arrays. It does work fine but ideally, the `message` and the `signature` should be hex strings too.